client-app/src/Components/DonorList.tsx

```diff
@@ -0,0 +1,159 @@
+                item.firstName.toLowerCase().includes(searchInput.toLowerCase()) ||
+                item.lastName.toLowerCase().includes(searchInput.toLowerCase()) ||
+                item.email.includes(searchInput)
+        );
+        setFilteredDonors(filtered);
+    };
+
+    const handleAddNewDonorClick = () => {
+        navigate('/donorform');
+    };
+
+    const handleViewDetailsClick = (donor: Donor) => {
+        selectedDonorDetails(donor);
+        setModalIsOpen(true);
+    };
+
+    return (
+        <div>
+            <div className="header">
+                <div className="logo-container">
+                    <img
+                        src="https://www.bworks.org/wp-content/themes/bworks/library/images/logo-bworks.png"
+                        alt="BWorks Logo"
+                        className="logo"
+                    />
+                </div>
+                <div className="options">
+                    <div className="search-bar">
+                        <input
+                            type="text"
+                            placeholder="Search for donor"
+                            value={searchInput}
+                            onChange={e => setSearchInput(e.target.value)}
+                        />
+                        <button className="search-button" onClick={handleSearch}>
+                            <FaSearch />
+                        </button>
+                    </div>
+                </div>
+            </div>
+
+            {error && <p className="error-message">{error}</p>}
+
+            <div className="div-topAddDonor">
+                <button onClick={handleAddNewDonorClick}>
+                    Add New Donor
+                </button>
+            </div>
+            
+            <table className="donor-list">
+                <thead>
+                    <tr>
+                        <th>Donor ID</th>
+                        <th>First Name</th>
+                        <th>Last Name</th>
+                        <th>Email</th>
+                        <th>More Details</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {(filteredDonors.length > 0 ? filteredDonors : currentDonors).map((donor, index) => (
+                        <tr key={donor.id}>
+                            <td>{donor.id}</td>
+                            <td>{donor.firstName}</td>
+                            <td>{donor.lastName}</td>
+                            <td>{donor.email}</td>
+                            <td>
+                                <button onClick={() => handleViewDetailsClick(donor)}>
+                                    View More Details
+                                </button>
+                            </td>
+                        </tr>
+                    ))}
+                </tbody>
+            </table>
+
+            <Modal isOpen={modalIsOpen} onRequestClose={() => setModalIsOpen(false)}>
```

Comment:
@truffer11 Please remove this modal. We do not need donor details in program page